python/ray/data/read_api.py

```diff
@@ -249,6 +259,7 @@ def read_datasource(
             prepare_read.remote(
                 datasource,
                 ctx,
+                cur_pg,
                 parallelism,
                 _wrap_and_register_arrow_serialization_workaround(read_args),
             )
```

Comment:
The check below needs to update (with the change to 1 CPU per task now etc.).
Also the warning log seems a bit off: it not necessarily has len(read_tasks) blocks since a read task can produce more than one blocks.